version: "3.9"
services:
  superduper:
    build:
      context: .
      target: dev
    network_mode: service:prosody
    volumes:
      - ./slidge:/venv/lib/python3.11/site-packages/slidge
      - ./superduper:/venv/lib/python3.11/site-packages/legacy_module
      - ./dev/assets:/venv/lib/python3.11/site-packages/legacy_module/assets
      - ./persistent:/var/lib/slidge
      - ./persistent/web:/slidge-web
      - ./dev/confs/slidge-example.ini:/etc/slidge/conf.d/slidge-example.ini
    depends_on:
      - prosody

  prosody:
    image: docker.io/nicocool84/slidge-prosody-dev:latest
    ports:
      - "127.0.0.1:5281:5281"  # XMPP port for clients to connect to
      - "127.0.0.1:5222:5222"  # prosody's http_file_share
      - "127.0.0.1:4444:4444"  # for nginx (optional, no-upload)
      - "127.0.0.1:8888:80"    # for movim (optional)

  nginx:
    image: docker.io/library/nginx:mainline-alpine
    volumes:
      - ./persistent/movim/html:/var/www/html:ro
      - ./dev/confs/nginx.conf:/etc/nginx/conf.d/default.conf:ro
      - ./persistent/web:/slidge-web
    network_mode: service:prosody

  postgresql:
    environment:
      POSTGRES_DB: movim
      POSTGRES_PASSWORD: BAD
      POSTGRES_USER: movim
    image: docker.io/library/postgres:14-alpine
    volumes:
      - ./persistent/movim/postgres/data:/var/lib/postgresql/data:rw
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5

  movim:
    environment:
      POSTGRES_DB: movim
      POSTGRES_HOST: postgresql
      POSTGRES_PORT: 5432
      POSTGRES_USER: movim
      POSTGRES_PASSWORD: BAD
    image: docker.io/movim/movim:0.21rc6
    network_mode: service:prosody
    volumes:
      - ./persistent/movim/html:/var/www/html:rw
      - ./dev/confs/movim.env:/var/www/html/.env:ro
    depends_on:
      postgresql:
        condition: service_healthy
    command:
      - su
      - -s
      - /bin/sh
      - -c
      - php daemon.php start
      - www-data
